### PR TITLE
Miscellaneous README changes

### PR DESCRIPTION
**Pull request by iKevinY**
- Various grammatical changes.
- Use triple-backtick syntax and `sh` highlighting for code blocks.
- Fix indentation of code block in step 2 of "Building on Windows".
- Use title case for "Getting Help" subheading.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -15,28 +15,34 @@ Read ["Installing Rust"] from [The Book].
...
+   * `git`
 
 2. Clone the [source] with `git`:
 
-        $ git clone https://github.com/rust-lang/rust.git
-        $ cd rust
+   ```sh
+   $ git clone https://github.com/rust-lang/rust.git
+   $ cd rust
+   ```
```
Comment (steveklabnik) [inline README.md]: if we're using triple graves, then we shouldn't need to indent

File: README.md [hunk 2]
```diff
@@ -15,28 +15,34 @@ Read ["Installing Rust"] from [The Book].
...
 
-    > ***Note:*** You may need to use `sudo make install` if you do not normally have
-    > permission to modify the destination directory. The install locations can
-    > be adjusted by passing a `--prefix` argument to `configure`. Various other
-    > options are also supported, pass `--help` for more information on them.
+    > ***Note:*** You may need to use `sudo make install` if you do not
+    > normally have permission to modify the destination directory. The
+    > install locations can be adjusted by passing a `--prefix` argument
+    > to `configure`. Various other options are also supported – pass
+    > `--help` for more information on them.
```
Comment (steveklabnik) [inline README.md]: same here, might as well outdent this too

**Comment by steveklabnik** [comment]
r= me after a few nits, thanks!

**Comment by iKevinY** [comment]
@steveklabnik I think the indentation is still necessary in order to keep the code blocks / blockquotes on the same "level" as the ordered lists (unless the content after the code block in step 3 of "Building From Source" should be excluded from the list). The list numberings get messed up if a top-level element appears in the middle of list, like the "Building on Windows" list on the current version of the README.

**Comment by steveklabnik** [comment]
Ahhhhh

**Comment by iKevinY** [comment]
That being said, maybe the paragraph after the note in "Building from Source" should be placed at the top level, since it's more of an informative statement rather than a "step" to follow.

**Pull request merged**